courses/api.py

```diff
@@ -258,11 +258,21 @@ def create_run_enrollments(
                 run=run,
                 defaults=dict(edx_enrolled=edx_request_success, enrollment_mode=mode),
             )
-            if not created and not enrollment.active:
-                enrollment.edx_enrolled = edx_request_success
-                subscribed = subscribe_to_edx_course_emails(user, run)
-                enrollment.edx_emails_subscription = True if subscribed else False
```

Comment:
I've moved this to a celery task, We aren't doing anything extra if the result of this API is true/false that needs to be here. So we can let celery task do this and continue with enrollment.